pkg/skaffold/build/cache/lookup.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2019 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cache
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/tag"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest"
+	"github.com/pkg/errors"
+)
+
+var (
+	// For testing
+	hashForArtifact = getHashForArtifact
+)
+
+func (c *cache) lookupArtifacts(ctx context.Context, tags tag.ImageTags, artifacts []*latest.Artifact) []cacheDetails {
+	details := make([]cacheDetails, len(artifacts))
+
+	var wg sync.WaitGroup
+	for i := range artifacts {
+		wg.Add(1)
+
+		i := i
```

Comment:
really just asking for my own curiosity: is there any advantage to reassigning the variable here as opposed to just passing it to the goroutine as a parameter?

Reply:
same same